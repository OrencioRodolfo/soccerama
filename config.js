export const api = { // eslint-disable-line
  endpoint: 'https://soccer.sportmonks.com/api',
  version: 'v2.0',
  token: 'QZgI6doracDdn2TIF8Rlc6ncZJkKqYK8ehUcogMeKFXWDTLReC6r8hUoWEva',
}
