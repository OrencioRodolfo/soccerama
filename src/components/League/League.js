import React, { Component } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import Standing from '../Standing/Standing'
import actions from '../../store/actions'
import TeamDialog from '../Team/Team'

const Title = styled.h1`
  text-align: center;
`

class League extends Component {
  static propTypes = {
    getLeague: PropTypes.func.isRequired,
    getStandings: PropTypes.func.isRequired,
    getSeason: PropTypes.func.isRequired,
    getTeam: PropTypes.func.isRequired,
    league: PropTypes.shape({
      id: PropTypes.number,
      name: PropTypes.string,
      currentSeasonId: PropTypes.number,
    }).isRequired,
    season: PropTypes.shape({
      id: PropTypes.number,
      name: PropTypes.string,
    }).isRequired,
    team: PropTypes.shape({
      id: PropTypes.number,
      name: PropTypes.string,
      logo: PropTypes.string,
    }).isRequired,
    squad: PropTypes.arrayOf(PropTypes.shape({
      id: PropTypes.number,
      name: PropTypes.string,
      logo: PropTypes.string,
    })).isRequired,
    standings: PropTypes.arrayOf(PropTypes.shape({
      id: PropTypes.number.isRequired,
      position: PropTypes.number.isRequired,
      name: PropTypes.string.isRequired,
      played: PropTypes.number.isRequired,
      won: PropTypes.number.isRequired,
      draw: PropTypes.number.isRequired,
      lost: PropTypes.number.isRequired,
      goal: PropTypes.number.isRequired,
      difference: PropTypes.string.isRequired,
      points: PropTypes.number.isRequired,
    })).isRequired,
  }

  state = {
    queryTeam: false,
  }

  componentDidMount() {
    this.fetchData(501)
  }

  fetchData = (id) => {
    this.props.getLeague(id).then((res) => {
      this.props.getStandings(res.value.currentSeasonId)
      this.props.getSeason(res.value.currentSeasonId)
    })
  }

  queryTeam = id => this.props.getTeam(id).then(() => {
    this.setState({
      queryTeam: true,
    })
  })

  handleTeamDailogClose = () => {
    this.setState({
      queryTeam: false,
    })
  }

  render() {
    const { league, season } = this.props
    return (
      <div>
        {league.name && season.name && <Title>{league.name} - {season.name}</Title>}
        <TeamDialog
          open={this.state.queryTeam}
          team={this.props.team}
          squad={this.props.squad}
          onClose={this.handleTeamDailogClose}
        />
        <Standing
          standings={this.props.standings}
          onTeamSelect={this.queryTeam}
        />
      </div>
    )
  }
}

const mapStateToProps = ({ league, team }) => ({
  league: league.details,
  season: league.season,
  team: team.details,
  squad: team.squad,
  standings: league.standings.map((item, index) => ({
    id: item.team_id,
    position: (index + 1),
    name: item.team_name,
    played: item.overall.games_played,
    won: item.overall.won,
    draw: item.overall.draw,
    lost: item.overall.lost,
    goal: item.overall.goals_scored,
    difference: item.total.goal_difference,
    points: item.points,
  })),
})

export default connect(mapStateToProps, {
  getLeague: actions.getLeague,
  getStandings: actions.getStandings,
  getSeason: actions.getSeason,
  getTeam: actions.getTeam,
})(League)

export {
  League as DLeague,
  mapStateToProps,
}
