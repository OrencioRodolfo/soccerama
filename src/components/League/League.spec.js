import React from 'react'
import { shallow } from 'enzyme'
import axios from 'axios'
import MockAdapter from 'axios-mock-adapter'
import { DLeague, mapStateToProps } from './League'
import { api } from '../../../config'

describe('Home component', () => {
  let mock
  let baseUrl
  let getLeague
  let getStandings
  let getSeason
  let getTeam
  let league
  let season
  let team
  let squad
  let standings
  let wrapper

  beforeEach(() => {
    mock = new MockAdapter(axios)
    baseUrl = `${api.endpoint}/${api.version}`
    const player = {
      id: 1,
      name: 'sample',
      logo: 'logo',
    }
    const standing = {
      id: 1,
      position: 2,
      name: 'sample',
      played: 3,
      won: 4,
      draw: 5,
      lost: 6,
      goal: 7,
      difference: '+8',
      points: 9,
    }

    league = {
      id: 1,
      name: 'sample',
      currentSeasonId: 2,
    }
    season = {
      id: 1,
      name: 'sample',
    }
    team = {
      id: 1,
      name: 'sample',
      logo: 'logo',
    }
    squad = [player, { ...player, id: 2 }]
    standings = [standing, { ...standing, id: 2 }]
    getLeague = () => Promise.resolve({ value: standings })
    getStandings = () => Promise.resolve(standings)
    getSeason = () => Promise.resolve(season)
    getTeam = () => Promise.resolve(team)

    wrapper = shallow(<DLeague
      getLeague={getLeague}
      getStandings={getStandings}
      getSeason={getSeason}
      getTeam={getTeam}
      league={league}
      season={season}
      team={team}
      squad={squad}
      standings={standings}
    />)
  })

  it('must render properly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('must fetch the team details', async () => {
    const teamId = 1
    mock.onGet(`${baseUrl}/teams/${teamId}?include=squad`).reply(200, {})
    await wrapper.instance().queryTeam(teamId).then(() => {
      expect(wrapper.instance().state.queryTeam).toEqual(true)
    })
  })

  it('must reset the "queryTeam" state when closed', () => {
    wrapper.instance().handleTeamDailogClose()
    expect(wrapper.instance().state.queryTeam).toEqual(false)
  })

  describe('mapStateToProps function', () => {
    it('must return the data related to the season, team, squad and standings', () => {
      const state = {
        league: {
          details: {},
          season: {},
          standings: [{
            team_id: 1,
            team_name: 'sample',
            overall: {},
            total: {},
            points: 10,
          }],
        },
        team: {
          details: {},
          squad: [],
        },
      }
      const result = mapStateToProps(state)
      expect(result).toHaveProperty('team')
      expect(result).toHaveProperty('squad')
      expect(result).toHaveProperty('standings')
      expect(result).toMatchSnapshot()
    })
  })
})
