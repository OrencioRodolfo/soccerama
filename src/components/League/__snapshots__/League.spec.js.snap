// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component mapStateToProps function must return the data related to the league, season, team, squad and standings 1`] = `
Object {
  "league": Object {},
  "season": Object {},
  "squad": Array [],
  "standings": Array [
    Object {
      "difference": undefined,
      "draw": undefined,
      "goal": undefined,
      "id": 1,
      "lost": undefined,
      "name": "sample",
      "played": undefined,
      "points": 10,
      "position": 1,
      "won": undefined,
    },
  ],
  "team": Object {},
}
`;

exports[`Home component must render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <League
    getLeague={[Function]}
    getSeason={[Function]}
    getStandings={[Function]}
    getTeam={[Function]}
    league={
      Object {
        "currentSeasonId": 2,
        "id": 1,
        "name": "sample",
      }
    }
    season={
      Object {
        "id": 1,
        "name": "sample",
      }
    }
    squad={
      Array [
        Object {
          "id": 1,
          "logo": "logo",
          "name": "sample",
        },
        Object {
          "id": 2,
          "logo": "logo",
          "name": "sample",
        },
      ]
    }
    standings={
      Array [
        Object {
          "difference": "+8",
          "draw": 5,
          "goal": 7,
          "id": 1,
          "lost": 6,
          "name": "sample",
          "played": 3,
          "points": 9,
          "position": 2,
          "won": 4,
        },
        Object {
          "difference": "+8",
          "draw": 5,
          "goal": 7,
          "id": 2,
          "lost": 6,
          "name": "sample",
          "played": 3,
          "points": 9,
          "position": 2,
          "won": 4,
        },
      ]
    }
    team={
      Object {
        "id": 1,
        "logo": "logo",
        "name": "sample",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <styled.h1>
          sample
           - 
          sample
        </styled.h1>,
        <Connect(WithTheme(WithWidth(WithMobileDialog)))
          onClose={[Function]}
          open={false}
          squad={
            Array [
              Object {
                "id": 1,
                "logo": "logo",
                "name": "sample",
              },
              Object {
                "id": 2,
                "logo": "logo",
                "name": "sample",
              },
            ]
          }
          team={
            Object {
              "id": 1,
              "logo": "logo",
              "name": "sample",
            }
          }
        />,
        <Standing
          onTeamSelect={[Function]}
          standings={
            Array [
              Object {
                "difference": "+8",
                "draw": 5,
                "goal": 7,
                "id": 1,
                "lost": 6,
                "name": "sample",
                "played": 3,
                "points": 9,
                "position": 2,
                "won": 4,
              },
              Object {
                "difference": "+8",
                "draw": 5,
                "goal": 7,
                "id": 2,
                "lost": 6,
                "name": "sample",
                "played": 3,
                "points": 9,
                "position": 2,
                "won": 4,
              },
            ]
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            "sample",
            " - ",
            "sample",
          ],
        },
        "ref": null,
        "rendered": Array [
          "sample",
          " - ",
          "sample",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onClose": [Function],
          "open": false,
          "squad": Array [
            Object {
              "id": 1,
              "logo": "logo",
              "name": "sample",
            },
            Object {
              "id": 2,
              "logo": "logo",
              "name": "sample",
            },
          ],
          "team": Object {
            "id": 1,
            "logo": "logo",
            "name": "sample",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onTeamSelect": [Function],
          "standings": Array [
            Object {
              "difference": "+8",
              "draw": 5,
              "goal": 7,
              "id": 1,
              "lost": 6,
              "name": "sample",
              "played": 3,
              "points": 9,
              "position": 2,
              "won": 4,
            },
            Object {
              "difference": "+8",
              "draw": 5,
              "goal": 7,
              "id": 2,
              "lost": 6,
              "name": "sample",
              "played": 3,
              "points": 9,
              "position": 2,
              "won": 4,
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <styled.h1>
            sample
             - 
            sample
          </styled.h1>,
          <Connect(WithTheme(WithWidth(WithMobileDialog)))
            onClose={[Function]}
            open={false}
            squad={
              Array [
                Object {
                  "id": 1,
                  "logo": "logo",
                  "name": "sample",
                },
                Object {
                  "id": 2,
                  "logo": "logo",
                  "name": "sample",
                },
              ]
            }
            team={
              Object {
                "id": 1,
                "logo": "logo",
                "name": "sample",
              }
            }
          />,
          <Standing
            onTeamSelect={[Function]}
            standings={
              Array [
                Object {
                  "difference": "+8",
                  "draw": 5,
                  "goal": 7,
                  "id": 1,
                  "lost": 6,
                  "name": "sample",
                  "played": 3,
                  "points": 9,
                  "position": 2,
                  "won": 4,
                },
                Object {
                  "difference": "+8",
                  "draw": 5,
                  "goal": 7,
                  "id": 2,
                  "lost": 6,
                  "name": "sample",
                  "played": 3,
                  "points": 9,
                  "position": 2,
                  "won": 4,
                },
              ]
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "sample",
              " - ",
              "sample",
            ],
          },
          "ref": null,
          "rendered": Array [
            "sample",
            " - ",
            "sample",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onClose": [Function],
            "open": false,
            "squad": Array [
              Object {
                "id": 1,
                "logo": "logo",
                "name": "sample",
              },
              Object {
                "id": 2,
                "logo": "logo",
                "name": "sample",
              },
            ],
            "team": Object {
              "id": 1,
              "logo": "logo",
              "name": "sample",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onTeamSelect": [Function],
            "standings": Array [
              Object {
                "difference": "+8",
                "draw": 5,
                "goal": 7,
                "id": 1,
                "lost": 6,
                "name": "sample",
                "played": 3,
                "points": 9,
                "position": 2,
                "won": 4,
              },
              Object {
                "difference": "+8",
                "draw": 5,
                "goal": 7,
                "id": 2,
                "lost": 6,
                "name": "sample",
                "played": 3,
                "points": 9,
                "position": 2,
                "won": 4,
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
